### ISSUE TITLE
Urdu Encoding Issue in Hugging Face Tokenizer 

### ISSUE THREAD

**Issue by El-chapo-007** [huggingface/transformers#30636]
### System Info

@ArthurZucker   i was training hugging face tokenizer on urdu dataset 
 However instead of having urdu tokens and merges in vocabulary it shows wiered encoding Here is an example:

ÚĪÛĮ': 393,
 'ĠØ³': 88,
 'ÙĪØ±': 101,
 'Ø¨Ø±': 255,
 'ĠÙħØª': 298,
 'ĠÙĪØ§ÙĦ': 249,
 'Ø³Ù¹': 305,
 'ĠÚ¯Ø§': 319,
 'ĺ': 61,
 'ĠØ¯ÙĨÛĮØ§': 324,
 '²': 20,
 'ĠØ¢': 146,
 'Ø±ÛĮ': 119,
 'Ø·': 174,
 'Ĺ': 60,
 'Ú©Ú¾': 335,
 'Ĵ': 55,
 'Ø§': 69,

### Who can help?

@ArthurZucker

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

from datasets import load_dataset

dataset = load_dataset("El-chapoo/Symboleess-urdu-1", split="train")

def get_training_corpus():
    for i in range(0, len(dataset), 1000):
        yield dataset[i : i + 1000]["cleaned_text"]

!pip install datasets
from datasets import load_dataset
from tokenizers import Tokenizer
from tokenizers.models import BPE
from transformers import AutoTokenizer

from tokenizers import (
    decoders,
    models,
    normalizers,
    pre_tokenizers,
    processors,
    trainers,
    Tokenizer,
)

tokenizer = Tokenizer(models.BPE())

tokenizer.pre_tokenizer = pre_tokenizers.ByteLevel(add_prefix_space=False)

trainer = trainers.BpeTrainer(vocab_size=400,special_tokens=["<pad>","<s>","</s>","<mask>"],
                          max_token_length=16)

tokenizer.train_from_iterator(get_training_corpus(), trainer=trainer)

tokenizer.get_vocab()

### Expected behavior

I have followed the guidelines from hugging face documentation . However gpt2  tokenizer also show same behaviour ,But Llama and Gemma tokenizer work expected but the consume alot of memory and requires much more time when training

**Comment by El-chapo-007** [comment]
when we save it the issue is solved
